jax/random.py

```diff
@@ -1349,3 +1349,133 @@ def _t(key, df, shape, dtype):
   half_df = lax.div(df, two)
   g = gamma(key_n, half_df, shape, dtype)
   return n * jnp.sqrt(half_df / g)
+
+
+def rademacher(key, shape, dtype=dtypes.int_):
+  """Sample from a Rademacher distribution.
+
+  Args:
+    key: a PRNGKey key.
+    shape: The shape of the returned samples.
+    dtype: The type used for samples.
+
+  Returns:
+    A jnp.array of samples, of shape `shape`. Each element in the output has
+    a 50% change of being 1 or -1.
+
+  """
+  dtype = dtypes.canonicalize_dtype(dtype)
+  shape = abstract_arrays.canonicalize_shape(shape)
+  return _rademacher(key, shape, dtype)
+
+
+@partial(jit, static_argnums=(1, 2))
+def _rademacher(key, shape, dtype):
+  random_bernoulli = categorical(
+      key=key, logits=jnp.array([0.5, 0.5]), shape=shape)
```

Comment:
I suspect it would be more performant here to generate these using a uniform distribution, which is closer to the metal than categorical (which computes a gumbel distribution under the hood)

Reply:
Thank you! I have now switched to `bernoulli` since its implementation is extremely close to what you have suggested.